packages/slice-machine/src/legacy/lib/builders/CustomTypeBuilder/SliceZone/index.tsx

```diff
@@ -429,6 +433,38 @@ const SliceZone: React.FC<SliceZoneProps> = ({
       )}
       <GenerateSliceWithAiModal
         open={isGenerateSliceWithAiModalOpen}
+        onSuccess={async (slices: SharedSlice[]) => {
```

Comment:
the `CreateSliceModal` also has a `onSuccess` callback and does the same thing, but we're doing here more things than we're doing there, probably because `CreateSliceModal` partially does some of those things

ideally, we should be able to remove somethings from the `CreateSliceModal` and bring them here, and have a single function that can be called from both `onSuccess` implementations here to add the slices to the slice zone, that way we avoid complex duplication

Reply:
I've given this a go. Let me know if this is what you had in mind.